gen/enum.go

```diff
@@ -265,6 +275,34 @@ func enumItemName(enumName string, spec *compile.EnumItem) (string, error) {
 	return enumName + name, err
 }
 
+// enumItemLabelName returns the actual name used for serialization/deserialization
+// default to EnumItem.Name, override by the value of GoLabel
```

Comment:
nit:

    // enumItemLabelName returns the label we use for this enum item in the generated code.

(I'm trying to make it clear that the label actually has nothing to do with
serialization for the use cases that ThriftRW supports. Using ThriftRW objects
as JSON is technically unsupported; it just happens to work.)

Reply:
agreed, the serialization/deserialization is pretty misleading given ThriftRW's context here